bls12_381/src/pairings.rs

```diff
@@ -276,6 +287,157 @@ impl<'a, 'b> Mul<&'b Scalar> for &'a Gt {
 impl_binops_additive!(Gt, Gt);
 impl_binops_multiplicative!(Gt, Scalar);
 
+impl Sum for Gt {
+    fn sum<I: Iterator<Item = Self>>(iter: I) -> Self {
+        iter.fold(Self::identity(), Add::add)
+    }
+}
+
+impl<'r> Sum<&'r Gt> for Gt {
+    fn sum<I: Iterator<Item = &'r Self>>(iter: I) -> Self {
+        iter.fold(Self::identity(), Add::add)
+    }
+}
+
+impl Group for Gt {
+    type Scalar = Scalar;
+
+    fn random<R: RngCore + ?Sized>(rng: &mut R) -> Self {
+        Gt(Fp12::random(rng))
```

Comment:
Not all elements of Fp12 are elements of the prime order multiplicative subgroup.

Replace with:

`MillerLoopResult(Fp12::random(rng)).final_exponentiation()` or something like that, and ensure that Fp12::random() result is nonzero.